localstack-core/localstack/services/ec2/provider.py

```diff
@@ -560,7 +560,14 @@ def get_security_groups_for_vpc(
                 GroupName=sg.name,
                 OwnerId=context.account_id,
                 PrimaryVpcId=sg.vpc_id,
-                Tags=[{"Key": k, "Value": v} for k, v in (sg.get_tags() or {}).items()],
+                Tags=[
+                    {
+                        "Key": tag.get("key") or tag.get("Key"),
+                        "Value": tag.get("value") or tag.get("Value"),
+                    }
+                    for tag in (sg.get_tags() or [])
```

Comment:
`get_tags()` [returns a list](https://github.com/getmoto/moto/blob/master/moto/ec2/models/core.py#L9-L13) so the conditional is redundant

```suggestion
                    for tag in sg.get_tags()
```